### PR TITLE
boards: beaglev_fire: improve docs, fix uart, expand RAM

### LINKED ISSUE #79483: doc: Documentation issue in 'boards/beagle/beaglev_fire/doc/index'
Example for building "hello world" does not work as it is missing the board qualifier.

It should be: `west build -b beaglev_fire/<board qualifier> samples/hello_world`
 
Where `<board qualifier>` = `polarfire/E51  or polarfire/u54  or polarfire/u54/smp`
                                         
* Page: `boards/beagle/beaglev_fire/doc/index`
* Version: main
* SHA-1: 22df8459faf4709590750c8223873fab27fc9169

### PR DESCRIPTION
**Pull request by AlexCharlton**
This PR makes three small changes, all aimed at improving BeagleV-Fire support.

### Update documentation
Aside from some minor clarifications, a "Flashing" section was added, since the steps to flash the Fire in a way that's compatible with the Microchip-provided [Hart Software Services](https://github.com/polarfire-soc/hart-software-services) (HSS) is not readily accomplished with `west/make flash`. An example of west scripts that are able to flash the Fire can be seen [here](https://github.com/polarfire-soc/zephyr-applications/), but I doubt this ought to live in Zephyr proper.

### Change selected UART
In https://github.com/zephyrproject-rtos/zephyr/commit/166b9bf35acc3a6c8a3665c11fa6e9b7054506c0#diff-01f5d360c1043dce50bf74fbb1cc056d2049160741e483f843c19d081c065252R18 the selected UART was changed from `uart0` to `uart1`, presumably to mirror the changes made to the other PolarFire boards. While this makes sense to do for the Icicle kit, from looking at the [Fire's schematics](https://git.beagleboard.org/beaglev-fire/beaglev-fire/-/blob/main/BeagleV-Fire_sch.pdf), I don't believe any of the UARTs other than `uart0` is hooked up to anything. 

### Expand available RAM
`sram0` is a relatively small region of RAM, meant to be used by the HSS (see [Linux device tree](https://github.com/linux4sam/linux-at91/blob/bf96df830936986fdb3c7789749ff599300dae01/arch/riscv/boot/dts/microchip/mpfs-beaglev-fire.dts#L73)). I see no reason why `sram1` shouldn't occupy the remaining ~2GB present on the Fire. I'm a Zephyr newbie, however, so I'm not sure if there are any reasons why you would want to limit the RAM defined.

### CODE REVIEW

**Comment by AlexCharlton** [comment]
Noting that these doc fixes overlap with https://github.com/zephyrproject-rtos/zephyr/pull/79511. My doc additions do cover the flashing instructions which were noted as missing in that PR.

**Comment by con-pax** [comment]
Hey @AlexCharlton, thanks for the PR! It all looks good to me. I will test it on the Beagle today.
Cheers

**Comment by AlexCharlton** [comment]
Thanks @con-pax! I should have noted that I tested this out on my Fire, and it Works For Me 🙂

**Review by con-pax — DISMISSED** [review]
LGTM
```bash
6.406446] >> *** Booting Zephyr OS build v4.0.0-rc1-79-g464e68645d7f ***
Hello World! beaglev_fire/polarfire/e51
```

```bash
464e68645d7 (HEAD -> main, origin/main, origin/HEAD) boards: beaglev_fire: expand RAM
```

```bash
pc             0x80000290	0x80000290 <main+24>
```

File: boards/beagle/beaglev_fire/doc/index.rst [hunk 1]
```diff
@@ -15,11 +15,17 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
...
+* ``beaglev_fire_polarfire_e51``: Uses only the E51 core
+* ``beaglev_fire_polarfire_u54``: Uses the U54 cores
+* ``beaglev_fire_polarfire_u54_smp``: Uses the U54 cores with CONFIG_SMP=y
+
 Applications for the ``beaglev_fire`` board configuration can be built as usual:
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: beaglev_fire
+   :board: beaglev_fire_polarfire_u54
```
Comment (Ayush1325) [inline boards/beagle/beaglev_fire/doc/index.rst:28]: The board name should be `beaglev_fire/polarfire/u54`

Comment (AlexCharlton) [reply boards/beagle/beaglev_fire/doc/index.rst:28]: Yes! Good catch 🙏

Comment (AlexCharlton) [reply boards/beagle/beaglev_fire/doc/index.rst]: Fixed

**Review by Ayush1325 — APPROVED** [review]

**Review by con-pax — APPROVED** [review]

**Review by vaishnavachath — APPROVED** [review]

**Comment by kartben** [comment]
@jadonk we still don't have you properly set up as a maintainer, apparently...

**Comment by kartben** [comment]
rebased to re-trigger incomplete CI

**Pull request merged**